### PR TITLE
new RICH reconstruction including direct and reflected light reconstruction

### PR DESCRIPTION
**Pull request by raffaelladevita**
- new code structure
- constants are read from DB
- new banks names and IDs according to convention

### CODE REVIEW

**Comment by baltzell** [comment]
Do we really want to start naming banks "REC*" that are made by services other than EBTB/EBHB?  Or is the idea that these new REC banks will later be made by EB?

**Comment by raffaelladevita** [comment]
In the long term some of the new banks should in fact be made by the EB but when this will happen we’ll have to discuss the structure. For the immediate term, to avoid having the RICH service creating REC::… banks, the names in the hipo4 json have been changed to RICH::…. but the hipo3 json has not been updated. Should I do it?

File: reconstruction/rich/src/main/java/org/jlab/rec/rich/Quaternion.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		y *= sizeInv;
+		z *= sizeInv;
+		w *= sizeInv;
+	}
+	/** 
+	 * Multiplication between two Quaternions
+	 * @param qb the one that multiply
+	 * @param result of the multiplication 
+	 */
+	public void multiply(Quaternion qb, Quaternion result) {
```
Comment (baltzell) [inline reconstruction/rich/src/main/java/org/jlab/rec/rich/Quaternion.java]: Looks like this isn't threadsafe?

Comment (gavalian) [reply reconstruction/rich/src/main/java/org/jlab/rec/rich/Quaternion.java]: I think this is fine. It uses the this reference to access local data.

**Pull request merged**